Source/Extensions/Blazorise.TreeView/TreeView.razor.cs

```diff
@@ -58,6 +57,39 @@ await foreach ( var nodeState in paramNodes.ToNodeStates( HasChildNodesAsync, Ha
                 treeViewNodeStates.Add( nodeState );
             }
         }
+
+        if ( paramNodes is INotifyCollectionChanged observableCollection )
+        {
+            observableCollection.CollectionChanged -= OnCollectionChanged;
+            observableCollection.CollectionChanged += OnCollectionChanged;

```

Comment:
Why do you need to remove and add? Isn't there something more elegant for this?
Can we improve this to where it just does this work once?

Reply:
This still needs improvements. We need to figure out a way to unsubscribe events if the `observableCollection` reference changes.

Reply:
I guess you'd need to check the backing field always for the old collection vs the new :S.
You're saying an unlikely case where people start with an obervable collection but change it midway to non observable rght?

Reply:
I'm saying if a user creates a new list.

`<TreeView Nodes="myList">`

and

then

`myList = new ();`

Reply:
What if you always unsubscribe before `await base.SetParametersAsync( parameters );` is called and the collection is mutated with a different one altogether? Doesn't that fix that use case?

Reply:
🤔🤔... that's... actually a good idea.

Reply:
Updated